fbw-common/src/wasm/systems/systems/src/surveillance/egpws/electrical_harness.rs

```diff
@@ -0,0 +1,109 @@
+use crate::{
+    shared::LgciuGearExtension,
+    simulation::{InitContext, Read, SimulationElement, SimulatorReader, VariableIdentifier},
+    surveillance::{
+        egpws::runtime::EnhancedGroundProximityWarningComputerPinProgramming,
+        taws::TerrainAwarenessWarningSystemDiscreteInputs,
+    },
+};
+
+pub struct EgpwsElectricalHarness {
+    discrete_inputs: TerrainAwarenessWarningSystemDiscreteInputs,
+    pin_programs: EnhancedGroundProximityWarningComputerPinProgramming,
+
+    flaps_mode_off: bool,
+    landing_flaps_3: bool,
+
+    terr_off_id: VariableIdentifier,
+    sys_off_id: VariableIdentifier,
+    gs_mode_off_id: VariableIdentifier,
+    flap_mode_off_id: VariableIdentifier,
+    landing_flap_3_on_id: VariableIdentifier,
+    gs_cancel_self_test_id: VariableIdentifier,
+
+    sfcc_fap_1_id: VariableIdentifier,
+    sfcc_fap_5_id: VariableIdentifier,
+
+    ecam_cp_emer_canc_id: VariableIdentifier,
+
+    slew_active_id: VariableIdentifier,
+}
+
+impl EgpwsElectricalHarness {
+    pub(super) const GPWS_TERR_OFF_KEY: &str = "GPWS_TERR_OFF";
+    pub(super) const GPWS_SYS_OFF_KEY: &str = "GPWS_SYS_OFF";
+    pub(super) const GPWS_GS_OFF_KEY: &str = "GPWS_GS_OFF";
+    pub(super) const GPWS_FLAP_OFF_KEY: &str = "GPWS_FLAP_OFF";
+    pub(super) const GPWS_FLAPS3_KEY: &str = "GPWS_FLAPS3";
+    pub(super) const GPWS_TEST_KEY: &str = "GPWS_TEST";
+    pub(super) const SFCC_1_FAP_1_KEY: &str = "SFCC_1_FAP_1";
+    pub(super) const SFCC_1_FAP_5_KEY: &str = "SFCC_1_FAP_5";
+    pub(super) const ECP_DISCRETE_OUT_EMER_CANC_KEY: &str = "ECP_DISCRETE_OUT_EMER_CANC";
+    pub(super) const IS_SLEW_ACTIVE_KEY: &str = "IS SLEW ACTIVE";
+
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            discrete_inputs: TerrainAwarenessWarningSystemDiscreteInputs::default(),
+            pin_programs: EnhancedGroundProximityWarningComputerPinProgramming::default(),
```

Comment:
Just a note that we have in the guidelines that we should [avoid dependencies between child modules](https://github.com/flybywiresim/aircraft/blob/master/fbw-common/src/wasm/systems/guidelines.md#avoid-dependencies-between-child-modules).